src/Components/App/App.js

```diff
@@ -27,130 +29,181 @@ class App extends Component {
+      if (name === "music") {
         promise = await getAllMusic();
       }
-      if (name === 'podcasts') {
+      if (name === "podcasts") {
         promise = await getAllPodcasts();
       }
-      this.setState({[name]: promise})
+      this.setState({ [name]: promise });
     } catch (error) {
-      console.log(error)
+      console.log(error);
     }
+  };
+
+  resetState = ()=>{
+    this.setState(this.baseState)
+  }
+  setHistory = () =>{
+    this.setState({
+      history:[...this.state.history, this.state],
+    })
+  }
+  goBack = ()=>{
+    this.setState(this.state.history[this.state.history.length - 1])
   }
 
   updateActivityAnswers = (event) => {
-    if (event.target.id !== 'default') {
-      this.setState({[event.target.id]: [...this.state[event.target.id], event.target.textContent]})
+    if (event.target.id !== "default") {
+      this.setState({
+        [event.target.id]: [
+          ...this.state[event.target.id],
+          event.target.textContent,
+        ],
+      });
     }
-  }
+  };
 
   setActivities = (activities) => {
-    activities.forEach(activity => {
-      this.getActivityData(activity)
-    })
-    this.setState({activities: [...activities]})
-  }
+    activities.forEach((activity) => {
+      this.getActivityData(activity);
+    });
+    this.setState({ activities: [...activities] });
+  };
 
   determineRandomActivity = () => {
-    let allFilteredActivities = this.state.activities.reduce((filtered, activity) => {
-      if (activity.includes('games')) {
-        activity = activity.replace(/ games/gi, 'Games')
+    let allFilteredActivities = this.state.activities.reduce(
+      (filtered, activity) => {
+        if (activity.includes("games")) {
+          activity = activity.replace(/ games/gi, "Games");
+        }
+        let filteredActivities = this.filterActivity(
+          activity,
+          this.state[`${activity}Answers`]
+        );
+        filtered.push(filteredActivities);
+        return filtered;
+      },
+      []
+    );
+    let final = allFilteredActivities.flat();
+    console.log(final);
+    let randomNumber = Math.floor(Math.random() * final.length);
+    return final[randomNumber];
+  };
+
+  genreFilter = (activity, answers) => {
+    return this.state[activity].filter((singleActivity) => {
+      if (activity === "movies") {
+        return singleActivity.genre.some((genre) => answers.includes(genre));
       }
-      let filteredActivities = this.filterActivity(activity, this.state[`${activity}Answers`])
-      filtered.push(filteredActivities)
-      return filtered
-    }, [])
-    let final = allFilteredActivities.flat()
-    let randomNumber = Math.floor(Math.random() *final.length)
-    return final[randomNumber]
-  }
+      return answers.some((genre) => genre === singleActivity.genre);
+    });
+  };
+
+  ageFilter = (possibleSuggestions, answers) => {
+    let ageRestriction = answers.find((answer) => {
+      return answer.includes("0's");
```

Comment:
Curious about this one - is there a reason to use `0's` rather than `\'s` like we had before?  Wondering if there's a benefit for switching it!

Reply:
Yeah so `'s` was catching `that's fine` and breaking the filter for this functionality, as it has nothing to do with age. Since our finds are happening for all of our answers we had to be a little more specific

Reply:
Thanks!

Reply:
Good catch!